### DISCUSSION TITLE
[feature request] Support different `for` duration settings for normal state or error state

### DISCUSSION THREAD

**Discussion by SunRunAway** [grafana/grafana#73161] [Alerting]
I got an alerting message:

```
alertname    DatasourceError
datasource_uid    grafanacloud-prom
grafana_folder    xxxx-collections
ref_id    A
rulename    xxxx
Annotations
Error    [sse.dataQueryError] failed to execute query [A]: Get "https://xxxx-1.grafana.net/api/prom/api/v1/query?query=probe_success%7Btarget%3D%22http%3A%2F%2Flocalhost%3A6060%2Fhealthy%22%2C+instance%3D%22xxxx-host%22%7D&time=1691086380": EOF
```

It seems that the managed Prometheus (I use Grafana Cloud) having problems and can not serve at that time.

If we can't avoid this error because Prometheus error may rarely happen. Currently, my alert rule is set with a for duration of '0s'. However, is it better to change this behavior such that only when the alert is in an error state (errstate), the for duration is triggered and set to 120 seconds. In cases of a normal alert or when in a nodata_state, I would like the for duration to remain at 0 seconds. So we can avoid the alerting message at this situation.

Here's what I'm trying to achieve:

If the alert is in errstate, it should be pending for 60 seconds before being triggered. If the alert is in a normal state or nodata_state, it should be triggered immediately without any pending time.

For example, something like this:

for = '0s'
for_errstate = '120s'

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82551. Feel free to continue the discussion around this there. Thank you!
